pkg/proxy/iptables/proxier.go

```diff
@@ -1220,28 +1220,35 @@ func (proxier *Proxier) syncProxyRules() {
 		}
 
 		// Capture load-balancer ingress.
-		fwChain := svcInfo.firewallChainName
-		for _, ingress := range svcInfo.LoadBalancerIPStrings() {
-			if hasEndpoints {
-				// create service firewall chain
-				if chain, ok := existingNATChains[fwChain]; ok {
-					proxier.natChains.WriteBytes(chain)
-				} else {
-					proxier.natChains.Write(utiliptables.MakeChainLine(fwChain))
-				}
-				activeNATChains[fwChain] = true
-				// The service firewall rules are created based on ServiceSpec.loadBalancerSourceRanges field.
-				// This currently works for loadbalancers that preserves source ips.
-				// For loadbalancers which direct traffic to service NodePort, the firewall rules will not apply.
+		if len(svcInfo.LoadBalancerIPStrings()) > 0 && hasEndpoints {
+			// We could elide the firewall chain when loadBalancerSourceRanges
+			// is empty, but that adds code and makes the flow harder to reason
+			// about, for not much gain.
+			fwChain := svcInfo.firewallChainName
```

Comment:
It doesn't add _much_ code, and I was planning to bring that commit back...

Reply:
It's OK as a followup, I guess.

Reply:
I added a WIP commit, PTAL